pkg/client/legacyconnection/client.go

```diff
@@ -0,0 +1,443 @@
+/*
+Copyright Avast Software. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package legacyconnection
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/google/uuid"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/legacyconnection"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/mediator"
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/transport"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/kms"
+	"github.com/hyperledger/aries-framework-go/pkg/store/connection"
+	"github.com/hyperledger/aries-framework-go/spi/storage"
+)
+
+// InvitationMsgType defines the connection invite message type.
+const InvitationMsgType = legacyconnection.InvitationMsgType
+
+// ErrConnectionNotFound is returned when connection not found.
+var ErrConnectionNotFound = errors.New("connection not found")
+
+type options struct {
+	routerConnections  []string
+	routerConnectionID string
+}
+
+func applyOptions(args ...Opt) *options {
+	opts := &options{}
+
+	for i := range args {
+		args[i](opts)
+	}
+
+	return opts
+}
+
+// Opt represents option function.
+type Opt func(*options)
+
+// InvOpt represents option for the CreateInvitation function.
+type InvOpt Opt
+
+// WithRouterConnectionID allows you to specify the router connection ID.
+func WithRouterConnectionID(conn string) InvOpt {
+	return func(opts *options) {
+		opts.routerConnectionID = conn
+	}
+}
+
+// WithRouterConnections allows you to specify the router connections.
+func WithRouterConnections(conns ...string) Opt {
```

Comment:
what if we move all those helper methods like `WithRouterConnections` and `applyOptions` to separate file like `opts.go`? (also all `With..`s methods at the end of the file) 
I guess the same would go to `didexchange` and other clients, but still. It will reduce amount of code in one file.

Reply:
I think it will be better to do such kind of refactoring in another PR because `With` convention is used in a lot of places